src/components/application_manager/test/commands/mobile/add_command_request_test.cc

```diff
@@ -0,0 +1,195 @@
+    menu_params3[strings::position] = 30;
+    menu_params3[strings::menu_name] = "Iphone";
+    smart_objects::SmartObject msg_params3 = smart_objects::SmartObject(smart_objects::SmartType_Map);
+    msg_params3[strings::cmd_id] = 3;
+    msg_params3[strings::menu_params] = menu_params3;
+    msg_params3[strings::app_id] = 33;
+   (*msg3)[strings::msg_params] = msg_params3;
+    utils::SharedPtr<AddCommandRequest> req3 = CreateCommand<AddCommandRequest>(msg3);
+
+
+    CommandsMap commands_map;
+    commands_map.insert(CommandsMap::value_type(0u, req));
+    commands_map.insert(CommandsMap::value_type(0u, req2));
+    commands_map.insert(CommandsMap::value_type(0u, req3));
+
+    ON_CALL(app_mock, commands_map())
+        .WillByDefault(Return(DataAccessor<CommandsMap>(commands_map, lock_)));
+*/
+ /**********************************/
+    MockAppPtr mock_app = CreateMockApp();
+    ON_CALL(app_mngr_, application(_)).WillByDefault(Return(mock_app));
+    EXPECT_CALL(*mock_app, app_id());
+
+
+    MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+    (*msg)[strings::params][hmi_response::code] = 0;
+    //static_cast<hmi_apis::Common_Result::SUCCESS>(message[strings::params][hmi_response::code].asInt());
+    (*msg)[strings::params][strings::msg_params][strings::cmd_id] = 1;
+
+    application_manager::event_engine::Event event1(hmi_apis::FunctionID::UI_AddCommand);
+    event1.set_smart_object(msg);
+
+    //creating UI-AddCommand
+   /* smart_objects::SmartObjectSPtr ui_command = MessageHelper::CreateRequestObject(app_mngr_.GetNextHMICorrelationID());
+    (*ui_command)[strings::params][strings::function_id] =
+        static_cast<int>(hmi_apis::FunctionID::UI_AddCommand);
+    smart_objects::SmartObject menu_params_UI = smart_objects::SmartObject(smart_objects::SmartType_Map);
+    menu_params_UI[am::hmi_request::parent_id] = 3;
+    menu_params_UI[strings::position] = 30;
+    menu_params_UI[strings::menu_name] = "Iphone";
+    smart_objects::SmartObject msg_params_UI = smart_objects::SmartObject(smart_objects::SmartType_Map);
+    msg_params_UI[strings::cmd_id] = 3;
+    msg_params_UI[strings::menu_params] = menu_params_UI;
+    msg_params_UI[strings::app_id] = mock_app->app_id();
+   (*ui_command)[strings::msg_params] = msg_params_UI;
+
+    smart_objects::SmartObjectList requests = MessageHelper::CreateAddCommandRequestToHMI(mock_app, app_mngr_);
+    requests.push_back(ui_command);
+
+  application_manager::event_engine::Event event2(hmi_apis::Common_Result::SUCCESS);
+  event2.set_smart_object(req2);
+  EXPECT_NO_THROW(req2->on_event(event2));
+
+  application_manager::event_engine::Event event3(hmi_apis::Common_Result::SUCCESS);
+  event3.set_smart_object(req3);
+  EXPECT_NO_THROW(req3->on_event(event3));*/
+
+
+   EXPECT_CALL(*mock_app, UpdateHash());
+
+
+   EXPECT_NO_THROW(req1->on_event(event1));
+   /******************************************************************/
+
+}
```

Comment:
@wolfylambova22, are you really don't interested in message that this command is send to app manager?